### PR TITLE
NEW TcpKeepAlive reproducer, compile script

### PR DESCRIPTION
**Pull request by dstuchli**
This is the Tcpkeepalivesettings reproducer along with the compilation script for it. 

At the end of the compilation script, there is a part with checksums. My main question is: Should we consider using the checksums when uploading to artifactory? If yes, that means one more dependency (a requirement for OS) - an rpm package Perl-Digest-SHA for generating SHA-1 checksum. Or we will not bother with checksums on artifactory and ignore the errors it generates. It DOES NOT intervene with the proper functioning of the reproducer. Artifactory only warns you that it could not check the checksums. This question is also applicable to cli-netlite and dotnet-core. Thank you for the answer. I would rather ignore the artifactory errors/warns than install the new package.

### CODE REVIEW

**Comment by pematous** [comment]
> ..I would rather ignore the artifactory errors/warns than install the new package.

How we currently do with other artifacts that are uploaded to artifactory? ie.: JMS

**Comment by dstuchli** [comment]
> > ..I would rather ignore the artifactory errors/warns than install the new package.
> 
> How we currently do with other artifacts that are uploaded to artifactory? ie.: JMS

It looks like that the errors are almost everywhere. The checksums are correctly checked in a handful of artifacts

File: TcpKeepAliveSettings-reproducer/TcpKeepAliveSettings/TcpKeepAliveSettings.csproj [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="AMQPNetLite.Core" Version="2.3.0" />
```
Comment (rkubis) [inline TcpKeepAliveSettings-reproducer/TcpKeepAliveSettings/TcpKeepAliveSettings.csproj]: I am not sure if this is right. I think that this line will have effect of downloading and using upstream version of library. You should probably change reference to absolute/relative path of our DLLs from clients zip.

File: compile-reproducer.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+if [[ "$CLI_BUILD" != "" ]]; then
+	CLI_BUILD=".$CLI_BUILD"
+fi
+
+# Create directory for finished product
+rm -rf $TARGET_DIR
+mkdir -p $TARGET_DIR
+
+# Search and replace Package AMQPNetLite.Core with AMQP.dll
+sed -i 's/PackageReference Include="AMQPNetLite.Core" Version="2.3.0/Reference Include="AMQP.dll/g' $REPRODUCER_DIR/TcpKeepAliveSettings.csproj
```
Comment (rkubis) [inline compile-reproducer.sh]: OK, I see that you do change here which relates to my comment above.

File: compile-reproducer.sh [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+# Download dotnet core client
+if ! wget -q -t 10 "http://download-node-02.eng.bos.redhat.com/devel/candidates/amq/AMQ-CLIENTS-$CLI_VER$CLI_BUILD/amq-clients-$CLI_VER-dotnet-core.zip"; then
```
Comment (rkubis) [inline compile-reproducer.sh]: I am not sure if we should have this line here. This repo is public to the whole world and we use our (probably) internal path.

File: compile-reproducer.sh [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+cp $REPRODUCER_DIR/bin/Release/netcoreapp3.1/TcpKeepAliveSettings.runtimeconfig.json $TARGET_DIR/TcpKeepAliveSettings-reproducer-$CLI_VER
+cd $TARGET_DIR && zip -r TcpKeepAliveSettings-reproducer-$CLI_VER\.zip TcpKeepAliveSettings-reproducer-$CLI_VER
+
+# Generate checksums
+# ARTIFACT_MD5_CHECKSUM=$(md5sum TcpKeepAliveSettings-reproducer-2.9.0.zip | awk '{print $1}')
+# ARTIFACT_SHA1_CHECKSUM=$(shasum -a 1 TcpKeepAliveSettings-reproducer-2.9.0.zip | awk '{ print $1 }')
+
+# Upload to artifactory
+# curl --header "X-Checksum-MD5:${ARTIFACT_MD5_CHECKSUM}" --header "X-Checksum-Sha1:${ARTIFACT_SHA1_CHECKSUM}" -u anonymous: -X PUT "http://messaging-qe-repo.usersys.redhat.com:8081/artifactory/cli-netlite/reproducers/" -T "TcpKeepAliveSettings-reproducer-$CLI_VER.zip"
+curl -u anonymous: -X PUT "http://messaging-qe-repo.usersys.redhat.com:8081/artifactory/cli-netlite/reproducers/" -T "TcpKeepAliveSettings-reproducer-$CLI_VER.zip"
```
Comment (rkubis) [inline compile-reproducer.sh]: This repo is public!

File: compile-reproducer.sh (hunk 3 above)
Comment (dstuchli) [reply compile-reproducer.sh]: - [x] Remove wget from the build script, wget should in job itself

File: compile-reproducer.sh (hunk 4 above)
Comment (dstuchli) [reply compile-reproducer.sh]: That is true! So the upload has to be done by the job itself, thank you for pointing that out!

- [x] Remove curl command from the build script

File: compile-reproducer.sh (hunk 2 above)
Comment (dstuchli) [reply compile-reproducer.sh]: This actually leads me to the rhetorical question: Why am I using sed every time, why don't I just change it...

- [x] Change References inside the project file, remove sed command from the build script

**Comment by pematous** [comment]
Apart from valid @rkubis's comments, that need to be resolved, looks good.

**Comment by pematous** [comment]
> > > ..I would rather ignore the artifactory errors/warns than install the new package.
> > 
> > 
> > How we currently do with other artifacts that are uploaded to artifactory? ie.: JMS
> 
> It looks like that the errors are almost everywhere. The checksums are correctly checked in a handful of artifacts

I'd prefer to have the checksums in place, but I really don't insist on that, given we are already not doing so with other uploads.

**Comment by dstuchli** [comment]
> > > > ..I would rather ignore the artifactory errors/warns than install the new package.
> > > 
> > > 
> > > How we currently do with other artifacts that are uploaded to artifactory? ie.: JMS
> > 
> > 
> > It looks like that the errors are almost everywhere. The checksums are correctly checked in a handful of artifacts
> 
> I'd prefer to have the checksums in place, but I really don't insist on that, given we are already not doing so with other uploads.

Well, there is a need to install that Perl-Digest-SHA package...I am just wondering if it is worth it. But I think I will upload it with checksums as well

- [x] Add checksums generation commands (not to compile script)

**Review by rkubis — APPROVED** [review]
Looks good now, approved.

**Pull request merged**